content/user/email/advanced_email_troubleshooting.md

```diff
@@ -103,19 +103,22 @@ Check for a PTR record with the [intoDNS](https://intodns.com) tool
 
 ### SPF RECORDS
 
-An SPF record is yet another DNS record, this time a TXT record that allows you to specify which machines are ALLOWED to send mail for your domain. The set-up has many possibilities; do a web search to learn more about it.
+A Sender Policy Framework (SPF) record is yet another DNS record. This TXT record allows you to specify which machines are ALLOWED to send mail for your domain. For instance, if you use something like Constant Contact to send mail on behalf of your site, you will need the same records for Constant Contact that you have for your domain. The set-up has many possibilities; do a web search to learn more about it.
 
-It is easy to succumb to the temptation to make the SPF definition broader than it should be. Resist. Another point that is easy to overlook with SPF records is that the mail server that sends out the email for your site should be included in your SPF record. I'll repeat that. The mail server that sends out the email for your site MUST be included in your SPF record. If it is not, why are you doing this?
+It is easy to succumb to the temptation to make the SPF definition broader than it should be. Resist. Another point that is easy to overlook with SPF records is that the mail server that sends out the email for your site should be included in your SPF record. As mentioned before, any site or mail service that sends mail on behalf of your domain should have an SPF record to accurately identify you and any additional mail servers used to send mail from your site.
 
-Any, all or none of the above (DNS, MX, SPF, etc) may be set up for you by your domain registrar or hosting service. What is much more important is that you check, and fix anything that is not correct.
+Any, all or none of the above (DNS, MX, SPF, DKIM, or DMARC) may be set up for you by your domain registrar or hosting service. What is much more important is that you check, and fix anything that is not correct and ensure the ones that are helpful to your business are in place and correct.
 
-All of the above can be checked by visiting [http://www.dnsstuff.com](http://www.dnsstuff.com) and doing a dnsReport on your domain. Check the mail section of the report, and fix any reported warning, error or problem that is within your power.
+All of the above can be checked by visiting [https://mxtoolbox.com](https://mxtoolbox.com) and taking advantage of their various diagnostic tools.  The site opens with a form for your domain name and an "MX Lookup" button.  Entering your domain, and clicking the button will quickly let you know if something is incorrect.  From the result, you can also do a Blacklist Check and SMTP Test.
+
+A simple SPF record might look like:
+yoursite.com. 14400 IN TXT "v=spf1 ip4:###.###.###.### ~all" where the # signs represent the IP address of your mail server.
```

Comment:
I'll accept this, but I note: 
While `ip4:xxxxxxx` is valid, it can usually be replaced with `a`, since in most cases the A record is also the actual server. (An exception would be an advanced case like using CloudFlare with proxying enabled .... but in such a case I'd strongly recommend that the actual webserver NOT send any emails itself either, but instead outsource that to a reliable service whose main business is delivering email.